q2_assembly/_utils.py

```diff
@@ -167,3 +171,98 @@ def concatenate_files(input_files, output_file):
     """
     cmd = ["cat", *input_files]
     subprocess.run(cmd, stdout=open(output_file, "w"), check=True)
+
+
+def generate_shortuuid():
+    """
+    Generate a shortuuid string.
+
+    Returns: a shortuuid strings.
+    """
+    return shortuuid.uuid()
+
+
+def generate_uuid3(namespace, name):
+    """
+    Generate a UUID3 string.
+    Args:
+        - namespace: UUID namespace.
+        - name: Name to be used in the UUID generation.
+    Returns: UUID3 string.
+    """
+    return uuid.uuid3(namespace, name)
+
+
+def generate_uuid4():
+    """
+    Generate a UUID4 string.
+
+    Returns: UUID4 string.
+    """
+    return uuid.uuid4()
+
+
+def generate_uuid5(namespace, name):
+    """
+    Generate a UUID5 string.
+    Args:
+        - namespace: UUID namespace.
+        - name: Name to be used in the UUID generation.
+    Returns: UUID5 string.
+    """
+    return uuid.uuid5(namespace, name)
+
+
+def _generate_unique_uuid(uuid_func, namespace, contig_id, new_ids, uuid_type):
+    """Generates a unique UUID for a contig.
+    Args:
+        - uuid_func: UUID generation function.
+        - namespace: UUID namespace.
+        - contig_id: Contig ID, what will be changed.
+        - new_ids: Set of already generated UUIDs.
+        - uuid_type: the type of uuid choose by the user
+
+    Returns: the new generated contig_id
+    """
+    while True:
+        if uuid_type in ["uuid4", "shortuuid"]:
+            new_id = str(uuid_func())
+        else:
+            new_id = str(uuid_func(namespace, contig_id))
+
+        if new_id not in new_ids:
+            new_ids.add(new_id)
+            return new_id
+
+
+def modify_contig_ids(out, sample, uuid_type: str):
+    """Modifies the contig IDs to include the sample name and UUID.
+
+    Args:
+        out: output format
+        sample: Name of the sample to be processed.
+        uuid_type: UUID type to be used in the contig ID.
+    """
+    uuid_funcs = {
+        "uuid3": generate_uuid3,
+        "uuid4": generate_uuid4,
+        "uuid5": generate_uuid5,
+        "shortuuid": generate_shortuuid,
+    }
+    uuid_func = uuid_funcs.get(uuid_type)
+
+    path_to_contigs = os.path.join(str(out), f"{sample}_contigs.fa")
+    with tempfile.TemporaryDirectory() as tmp:
+        path_to_contigs_mod = os.path.join(tmp, f"{sample}_modified_contigs.fa")
+        namespace = uuid.NAMESPACE_OID
+        new_ids = set()
+        with io.open(path_to_contigs_mod, "w") as modified_contigs:
+            for contig in io.read(path_to_contigs, format="fasta"):
+                contig.metadata["id"] = _generate_unique_uuid(
+                    uuid_func, namespace, contig.metadata["id"], new_ids, uuid_type
+                )
+                io.write(contig, format="fasta", into=modified_contigs)
```

Comment:
This part would need to change to something like this:
```suggestion
        with io.open(path_to_contigs_mod, "w") as modified_contigs:
            for contig in io.read(path_to_contigs, format="fasta"):
                if uuid_type in ["uuid4", "shortuuid"]:
                    new_id = str(uuid_func())
                else:
                    new_id = str(uuid_func(NAMESPACE_OID, contig.metadata["id"]))

                contig.metadata["id"] = new_id
                io.write(contig, format="fasta", into=modified_contigs)
```